### PR TITLE
[15.0][IMP] stock_request_purchase: Propagate the cancellation of related purchases

### PR DESCRIPTION
**Pull request by victoralmau**
FWP from 14.0: https://github.com/OCA/stock-logistics-warehouse/pull/1710

Propagate the cancellation of related purchases

Please @pedrobaeza and @pilarvargas-tecnativa can you review it?

@Tecnativa TT43414

### CODE REVIEW

File: stock_request_purchase/models/stock_request.py [hunk 1]
```diff
@@ -38,6 +39,14 @@ def _check_purchase_company_constrains(self):
                 )
             )
 
+    def action_cancel(self):
+        """Propagate the cancellation to the generated purchase orders."""
+        res = super().action_cancel()
+        self.sudo().purchase_ids.filtered(
```
Comment (pedrobaeza) [inline stock_request_purchase/models/stock_request.py:45]: What happens if several products from other stock request/MTO/etc are in the same purchase order? I think you should check that the PO is only from this stock request.

**Review by pedrobaeza — APPROVED** [review]
Patch to be picked to v14

**Review by pilarvargas-tecnativa — APPROVED** [review]
thanks :)

**Pull request merged**